src/EventListener/DataContainer/Theme.php

```diff
@@ -0,0 +1,379 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2018 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\Backend;
+use Contao\BackendUser;
+use Contao\CoreBundle\Exception\AccessDeniedException;
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\FilesModel;
+use Contao\Image;
+use Contao\StringUtil;
+use Contao\StyleSheets;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class Theme
+{
+    /** @var RequestStack */
+    protected $requestStack;
+
+    /** @var SessionInterface */
+    protected $session;
+
+    /** @var BackendUser */
+    protected $user;
+
+    /** @var ImageFactoryInterface */
+    private $imageFactory;
+
+    /**
+     * Theme constructor.
+     *
+     * @param RequestStack          $requestStack
+     * @param SessionInterface      $session
+     * @param TokenStorageInterface $tokenStorage
+     * @param ImageFactoryInterface $imageFactory
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        SessionInterface $session,
+        TokenStorageInterface $tokenStorage,
+        ImageFactoryInterface $imageFactory
+    ) {
+        $this->requestStack = $requestStack;
+        $this->session      = $session;
+        $this->user         = $tokenStorage->getToken()->getUser();
+        $this->imageFactory = $imageFactory;
+    }
+
+    /**
+     * Check permissions to edit the table
+     *
+     * @throws AccessDeniedException
+     */
+    public function onCheckPermission(): void
+    {
+        if ($this->user->isAdmin) {
+            return;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
```

Comment:
Can be `null` e.g. on command line when building DCA cache etc.

Reply:
You're right! I'll add a check. I guess the same holds true for the user and I should better use `$user = $this->token->getUser();` in the functions?

Reply:
Yes :)